core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -1291,30 +1299,80 @@ OptimizelyDecision decide(@Nonnull OptimizelyUserContext user,
             flagEnabled,
             optimizelyJSON,
             ruleKey,
-            key,
+            flagKey,
             user,
             reasonsToReport);
     }
 
     Map<String, OptimizelyDecision> decideForKeys(@Nonnull OptimizelyUserContext user,
+                                                          @Nonnull List<String> keys,
+                                                          @Nonnull List<OptimizelyDecideOption> options) {
+        return decideForKeys(user, keys, options, false);
+    }
+
+    private Map<String, OptimizelyDecision> decideForKeys(@Nonnull OptimizelyUserContext user,
                                                   @Nonnull List<String> keys,
-                                                  @Nonnull List<OptimizelyDecideOption> options) {
+                                                  @Nonnull List<OptimizelyDecideOption> options,
+                                                  boolean ignoreDefaultOptions) {
```

Comment:
In this PR, decide() is implemented using decideForKeys() which does not return result for a flag if ENABLED_FLAGS_ONLY option is set. decideForKeys() adds the defualt options to the passed in options. In this implementation, decide() does this default options inclusion and then removes the ENABLED_FLAGS_ONLY option from the final list so that it can get a decision. If this flag is provided, decideForKeys() will skip the inclusion of defualt flag. 
Please note that this is a new overload of `decideForKeys` which is only used internally. The public interface in still unchanged.

Reply:
I see the point. With this approach, we may lose other default options like DISBLE_EVENT_DISPATCH, ... 
We can consider to change it to something like "oneFlagKey", so we can exclude "ENABLED_FLAGS_ONLY" only.